docs/reference/developer/notebook-style-guide.ipynb

```diff
@@ -0,0 +1,78 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "3205d1c4-8704-4b77-80b6-287e252864eb",
+   "metadata": {},
+   "source": [
+    "# Jupyter Notebook Style Guide\n",
+    "\n",
+    "## Motivation\n",
+    "\n",
+    "This style guide is meant for anyone writing documentation pages, tutorials, or workflows as Jupyter notebooks.\n",
+    "\n",
+    "The purpose is to make notebooks easy to understand and readable, and to minimize the reader's cognitive overload when switching between reading various notebooks.\n",
+    "\n",
+    "## Recommendations\n",
+    "\n",
+    "### Saving notebooks\n",
+    "\n",
+    "- Avoid storing outputs before saving or committing a notebook.\n",
+    "  - Use \"Restart kernel and clear all outputs\" before saving.\n",
+    "  - [nb-clean](https://github.com/srstevenson/nb-clean) can be used to do this automatically.\n",
+    "\n",
+    "### Document structure\n",
+    "\n",
+    "- Use markdown sections, subsections, ... for easy navigation and readability.\n",
+    "    \n",
+    "### Portability\n",
+    "\n",
+    "- Avoid hard-coding file paths which would require users to download files into specific folders.\n",
+    "- Use APIs, not file systems, to access data.\n",
+    "  - [pooch](https://www.fatiando.org/pooch/latest/) is a good candidate when sample files are needed.\n",
+    "\n",
+    "### Code formatting\n",
+    "\n",
+    "- Line lengths should be limited to avoid horizontal scroll bars when viewing documentation pages in a browser or in Jupyter.\n",
+    "  - A line length of 88 (default in `black`) seems to work well.\n",
+    "- Automatic code formatting (such as `black`, `yapf`, or `flake8`) is recommended, but in some cases it may hurt readability, so no binding recommendation is made, i.e., exceptions are allowed.\n",
+    "  - [jupyterlab-code-formatter](https://jupyterlab-code-formatter.readthedocs.io/en/latest/index.html) is a useful tool for applying automatic formatting to indivudual cells or entire notebooks. Example config for `black`:\n",
+    "    ```\n",
+    "    {\n",
+    "        {\n",
+    "        \"black\": {\n",
+    "            \"line_length\": 88,\n",
+    "            \"string_normalization\": false\n",
+    "        },\n",
+    "        \"preferences\": {\n",
+    "            \"default_formatter\": {\n",
+    "                \"python\": [\"black\"],\n",
+    "            }\n",
+    "        }\n",
+    "    }\n",
+    "    ```"
+   ]
+  }
+ ],
+ "metadata": {
+  "kernelspec": {
+   "display_name": "Python 3 (ipykernel)",
+   "language": "python",
+   "name": "python3"
+  },
+  "language_info": {
+   "codemirror_mode": {
+    "name": "ipython",
+    "version": 3
+   },
+   "file_extension": ".py",
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython3",
+   "version": "3.8.12"
+  }
+ },
+ "nbformat": 4,
+ "nbformat_minor": 5
+}
```

Comment:
Should this style guide live in `scipp` or in `ess` (or duplicated in all 3 repos??

Reply:
What we write here applies to scipp. I would say if we need more detailed guides, say in `ess`, we should refer to this as the base guide and add additional rules/recommendations there?